### PR TITLE
fix: slice reusability on db upload & SQL updates

### PR DESCRIPTION
**Pull request by cabreraalex**
# Description
Two main fixes:
* Always derive slice column ID from name and model to avoid issues on dataset updates
* Slight performance updates to histogram calculation

### CODE REVIEW

File: backend/zeno_backend/database/select.py [hunk 1]
```diff
@@ -1281,25 +1281,35 @@ def table_data(project: str, filter_sql: sql.Composed | None) -> SQLTable:
...
+def column_id_from_name_and_model(
+    project: str, column_name: str, model: str | None
+) -> str:
     """Get a column's id given its name and model.
 
     Args:
         project (str): the project the user is currently working with.
         column_name (str): the name of the column to be fetched.
-        model (str): the model of the column to be fetched.
+        model (str, optional): the model of the column to be fetched. Defaults to "".
```
Comment (Sparkier) [inline backend/zeno_backend/database/select.py:1292]: Does not default to "" rn.

File: backend/zeno_backend/database/select.py [hunk 2]
```diff
@@ -1281,25 +1281,35 @@ def table_data(project: str, filter_sql: sql.Composed | None) -> SQLTable:
...
-        sql.SQL(
-            "SELECT column_id FROM {} "
-            "WHERE name = %s AND (model = %s OR model IS NULL);"
-        ).format(sql.Identifier(f"{project}_column_map")),
-        [column_name, model],
-    )
+    if model is None:
+        column_result = db.connect_execute_return(
+            sql.SQL(
+                "SELECT column_id FROM {} " "WHERE name = %s AND model IS NULL;"
```
Comment (Sparkier) [inline backend/zeno_backend/database/select.py:1301]: nit: remove the `" "` in the middle of the query.

File: backend/zeno_backend/database/select.py [hunk 3]
```diff
@@ -1281,25 +1281,35 @@ def table_data(project: str, filter_sql: sql.Composed | None) -> SQLTable:
...
+            ).format(sql.Identifier(f"{project}_column_map")),
+            [column_name],
+        )
+    else:
+        column_result = db.connect_execute_return(
+            sql.SQL(
+                "SELECT column_id FROM {} " "WHERE name = %s AND model = %s;"
```
Comment (Sparkier) [inline backend/zeno_backend/database/select.py:1308]: nit: remove the `" "` in the middle of the query.

File: backend/zeno_backend/processing/histogram_processing.py [hunk 4]
```diff
@@ -40,16 +40,24 @@ async def histogram_bucket(project_uuid: str, col: ZenoColumn):
 
             if col.data_type == MetadataType.NOMINAL:
                 await db.execute(
-                    sql.SQL("SELECT {} FROM {} GROUP BY {}").format(
+                    sql.SQL(
+                        "SELECT COUNT(*) FROM (SELECT DISTINCT {} FROM {}) AS temp;"
```
Comment (Sparkier) [inline backend/zeno_backend/processing/histogram_processing.py:44]: nit: don't think you need the `AS temp` here, do you?

**Review by Sparkier — APPROVED** [review]

File: backend/zeno_backend/processing/histogram_processing.py (hunk 4 above)
Comment (cabreraalex) [reply backend/zeno_backend/processing/histogram_processing.py:44]: I *think* it speeds it up for some reason, google did some benchmarking.

**Pull request merged**